crates/hir/src/display.rs

```diff
@@ -595,6 +595,36 @@ impl HirDisplay for Trait {
         let def_id = GenericDefId::TraitId(self.id);
         write_generic_params(def_id, f)?;
         write_where_clause(def_id, f)?;
+
+        let assoc_items = self.items(f.db);
+        let assoc_items_size = assoc_items.len();
+        let max_display_size = f.max_size.unwrap_or(assoc_items_size);
```

Comment:
`max_size` is meant to limit the overall character count of a display, but the config here is supposed to limit by the number of assoc items. So we'll need to add a specific extra field to the formatter unfortunately that describes this limit and use that appropriately here.